packages/starlight/utils/format-path.ts

```diff
@@ -0,0 +1,47 @@
+import type { AstroConfig } from 'astro';
+import { fileWithBase, pathWithBase } from './base';
+import { stripExtension, ensureHtmlExtension } from './navigation';
+import { ensureTrailingSlash, stripTrailingSlash } from './path';
+
+interface FormatPathOptions {
+	format?: AstroConfig['build']['format'];
+	trailingSlash?: AstroConfig['trailingSlash'];
+}
+
+const formatStrategies = {
+	file: {
+		addBase: fileWithBase,
+		handleExtension: (href: string) => ensureHtmlExtension(href),
+	},
+	directory: {
+		addBase: pathWithBase,
+		handleExtension: (href: string) => stripExtension(href),
+	},
+};
+
+const trailingSlashStrategies = {
+	always: ensureTrailingSlash,
+	never: stripTrailingSlash,
+	ignore: (href: string) => href,
+};
+
+export function formatPath(href: string, opts?: FormatPathOptions) {
+	const options = {
+		format: opts?.format ?? 'directory',
+		trailingSlash: opts?.trailingSlash ?? 'ignore',
+	};
+
+	const formatStrategy = formatStrategies[options.format];
+	const trailingSlashStrategy = trailingSlashStrategies[options.trailingSlash];
```

Comment:
Nit, but in general for situations like this we use defaults in the signature. My preference is mainly based on avoiding the ambiguity of having both `opts` and `options` in scope. With this way of assigning the defaults you have a single option to consume inside the function body.

```suggestion
export function formatPath(
	href: string,
	{ format = 'directory', trailingSlash = 'ignore' }: FormatPathOptions = {}
) {
	const formatStrategy = formatStrategies[format];
	const trailingSlashStrategy = trailingSlashStrategies[trailingSlash];
```

This also switches from the nullish coalescing `??` logic to logic equivalent to `||`, but I don’t think we want to support stuff like `format: ''` or `trailingSlash: 0`, so should be equivalent?